undefined reference to clock_gettime()

i have followed the instruction described in READ ME file to build dataseries.
Finally after executing this comment :: perl ~/deptool-bootstrap build -t

OUTPUT::
Do you want to: missing $ENV{BUILD_ROOT}; assume /home/localuser/build
Return to continue, ctrl-c to abort: [auto-yes]

Do you want to: missing $ENV{BUILD_DEBUG}; assume /home/localuser/build/dbg-ubuntu-12.04-x86_64
Return to continue, ctrl-c to abort: [auto-yes]

Do you want to: missing $ENV{BUILD_OPT}; assume /home/localuser/build/opt-ubuntu-12.04-x86_64
Return to continue, ctrl-c to abort: [auto-yes]

Do you want to: missing $ENV{PROJECTS}; assume /home/localuser/projects
Return to continue, ctrl-c to abort: [auto-yes]
Up to date build in /home/localuser/build/opt-ubuntu-12.04-x86_64/Lintel
Up to date install in /home/localuser/build/opt-ubuntu-12.04-x86_64/Lintel
Up to date test in /home/localuser/build/opt-ubuntu-12.04-x86_64/Lintel
Start build in /home/localuser/build/opt-ubuntu-12.04-x86_64/DataSeries

---

ERROR: failed build in /home/localuser/build/opt-ubuntu-12.04-x86_64/DataSeries
-------------------- STDOUT:
depbuild: building...
Running: make -j 3
[ 28%] Built target DataSeries
[ 28%] Built target dstypes2cxx.1
[ 28%] Built target dataseries-config.1
Linking CXX executable csv2ds
[ 28%] Built target bacct2ds.1
Linking CXX executable ds2ellardnfs
[ 28%] Built target csv2ds.1
[ 28%] Built target ds2ellardnfs.1

-------------------- STDERR:
../libDataSeries.so.0.0: undefined reference to `clock_gettime'
collect2: ld returned 1 exit status
make[2]: *** [src/process/csv2ds] Error 1
make[1]: *** [src/process/CMakeFiles/csv2ds.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
../libDataSeries.so.0.0: undefined reference to`clock_gettime'
collect2: ld returned 1 exit status
make[2]: **\* [src/process/ds2ellardnfs] Error 1
make[1]: **\* [src/process/CMakeFiles/ds2ellardnfs.dir/all] Error 2
make: **\* [all] Error 2
'make -j 3' failed (512)
Failed on build in /home/localuser/build/opt-ubuntu-12.04-x86_64/DataSeries
Exiting on error, check previous output for precise problem.

What did I miss? Can you help me out?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,6 +45,7 @@ SET(CMAKE_MODULE_FILES
 	DataSeriesCMakeUtil.cmake
 	FindDataSeries.cmake
         FindThrift.cmake # TODO-reviewer: should this go into lintel?
+         FindLibRt.cmake
 )
 
 INSTALL(FILES ${CMAKE_MODULE_FILES}

--- a/DataSeriesCMakeUtil.cmake
+++ b/DataSeriesCMakeUtil.cmake
@@ -6,11 +6,12 @@
 # Utility macros for DataSeries
 
 INCLUDE(LintelDocs)
+INCLUDE(FindLibRt)
 
 MACRO(DATASERIES_PROGRAM_NOINST program_name)
     ADD_EXECUTABLE(${program_name} ${program_name}.cpp ${ARGN})
     ADD_DEPENDENCIES(${program_name} DataSeries)
-    TARGET_LINK_LIBRARIES(${program_name} DataSeries ${LINTEL_LIBRARIES})
+    TARGET_LINK_LIBRARIES(${program_name} DataSeries ${LIBRT_LIBRARIES} ${LINTEL_LIBRARIES})
 ENDMACRO(DATASERIES_PROGRAM_NOINST)
 
 # Generates a dataseries program, pass in the name of the executable; the assumption is that
@@ -19,7 +20,7 @@ ENDMACRO(DATASERIES_PROGRAM_NOINST)
 MACRO(DATASERIES_PROGRAM program_name)
     ADD_EXECUTABLE(${program_name} ${program_name}.cpp ${ARGN})
     ADD_DEPENDENCIES(${program_name} DataSeries)
-    TARGET_LINK_LIBRARIES(${program_name} DataSeries)
+    TARGET_LINK_LIBRARIES(${program_name} DataSeries ${LIBRT_LIBRARIES})
     INSTALL(TARGETS ${program_name} DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
     IF(DEFINED DATASERIES_POD2MAN_RELEASE)
         LINTEL_POD2MAN(${program_name}.cpp 1 ${DATASERIES_POD2MAN_RELEASE} "-" ${program_name}.1)

--- a/FindLibRt.cmake
+++ b/FindLibRt.cmake
@@ -1,0 +1,12 @@
+# Locate librt library
+# This module defines
+#  LIBRT_FOUND - System has librt
+#  LIBRT_LIBRARIES - The librt library
+
+find_library(LIBRT_LIBRARIES rt)
+
+include(FindPackageHandleStandardArgs)
+# Sets LIBRT_FOUND
+find_package_handle_standard_args(LibRt DEFAULT_MSG LIBRT_LIBRARIES)
+
+mark_as_advanced(LIBRT_LIBRARIES)

